scripts/validate_docstrings.py

```diff
@@ -517,42 +511,109 @@ def validate_one(func_name):
         if examples_errs:
             errs.append('Examples do not pass tests')
 
-    sys.stderr.write(_output_header('Validation'))
-    if errs:
-        sys.stderr.write('Errors found:\n')
-        for err in errs:
-            sys.stderr.write('\t{}\n'.format(err))
-    if wrns:
-        sys.stderr.write('Warnings found:\n')
-        for wrn in wrns:
-            sys.stderr.write('\t{}\n'.format(wrn))
+    return {'type': doc.type,
+            'docstring': doc.clean_doc,
+            'deprecated': doc.deprecated,
+            'file': doc.source_file_name,
+            'file_line': doc.source_file_def_line,
+            'github_link': doc.github_url,
+            'errors': errs,
+            'warnings': wrns,
+            'examples_errors': examples_errs}
+
 
-    if not errs:
-        sys.stderr.write('Docstring for "{}" correct. :)\n'.format(func_name))
+def validate_all():
```

Comment:
Assuming this to be the main function of the change I was thinking the test cases would cover this but don't see that - oversight on my end or something we can test?

Reply:
This function is just boilerplate code, I don't think it adds any value to test that, as we would be mocking everything.